realtime_test.go

```diff
@@ -170,6 +171,106 @@ func TestAlert(t *testing.T) {
 	}
 }
 
+func TestVehicle(t *testing.T) {
+	timestamp := uint64(time1.Unix())
+	header := &gtfsrt.FeedHeader{
+		GtfsRealtimeVersion: ptr("2.0"),
+		Timestamp:           &timestamp,
+	}
+	entities := []*gtfsrt.FeedEntity{
+		{
+			Id: ptr("1"),
+			Vehicle: &gtfsrt.VehiclePosition{
+				Trip: &gtfsrt.TripDescriptor{
+					TripId: ptr(tripID1),
+				},
+				Vehicle: &gtfsrt.VehicleDescriptor{
+					Id: ptr(vehicleID1),
+				},
+				Position: &gtfsrt.Position{
+					Latitude:  ptrf32(1.0),
+					Longitude: ptrf32(2.0),
+					Bearing:   ptrf32(3.0),
+					Odometer:  ptrf64(4.0),
+					Speed:     ptrf32(5.0),
+				},
+				CurrentStopSequence: ptru32(6),
+				StopId:              ptr(stopID1),
+				CurrentStatus:       ptrCurrentStatus(gtfsrt.VehiclePosition_STOPPED_AT),
+				Timestamp:           &timestamp,
+				CongestionLevel:     ptrCongestionLevel(gtfsrt.VehiclePosition_CONGESTION),
+				OccupancyStatus:     ptrOccupancyStatus(gtfsrt.VehiclePosition_EMPTY),
+			},
+		},
+	}
+
+	trip := gtfs.Trip{
+		ID: gtfs.TripID{
+			ID:          tripID1,
+			DirectionID: gtfs.DirectionIDUnspecified,
+		},
+		IsEntityInMessage: false,
+	}
+	position := gtfs.Position{
+		Latitude:  ptrf32(1.0),
+		Longitude: ptrf32(2.0),
+		Bearing:   ptrf32(3.0),
+		Odometer:  ptrf64(4.0),
+		Speed:     ptrf32(5.0),
+	}
+	vehicle := gtfs.Vehicle{
+		ID: &gtfs.VehicleID{
+			ID: ptr(vehicleID1),
+		},
+		Position:            &position,
+		CurrentStopSequence: ptru32(6),
+		StopID:              ptr(stopID1),
+		CurrentStatus:       ptrCurrentStatus(gtfsrt.VehiclePosition_STOPPED_AT),
+		Timestamp:           &time1,
+		CongestionLevel:     ptrCongestionLevel(gtfsrt.VehiclePosition_CONGESTION),
+		OccupancyStatus:     ptrOccupancyStatus(gtfsrt.VehiclePosition_EMPTY),
+		IsEntityInMessage:   true,
+	}
+	trip.Vehicle = &vehicle
+	vehicle.Trip = &trip
+
+	expectedResult := &gtfs.Realtime{
+		CreatedAt: time1,
+		Trips:     []gtfs.Trip{trip},
+		Vehicles:  []gtfs.Vehicle{vehicle},
+	}
+
+	result := testutil.MustParse(t, header, entities, &gtfs.ParseRealtimeOptions{})
+
+	if !reflect.DeepEqual(result, expectedResult) {
+		t.Errorf("actual:\n%+v\n!= expected:\n%+v", result, expectedResult)
+	}
+}
+
 func ptr(s string) *string {
 	return &s
 }
+
+func ptru32(u uint32) *uint32 {
+	return &u
+}
+
+func ptrf32(f float32) *float32 {
+	return &f
+}
+
+func ptrf64(f float64) *float64 {
+	return &f
+}
+
+func ptrCurrentStatus(s gtfsrt.VehiclePosition_VehicleStopStatus) *gtfsrt.VehiclePosition_VehicleStopStatus {
+	return &s
+}
+
+func ptrCongestionLevel(s gtfsrt.VehiclePosition_CongestionLevel) *gtfsrt.VehiclePosition_CongestionLevel {
+	return &s
+}
+
+func ptrOccupancyStatus(s gtfsrt.VehiclePosition_OccupancyStatus) *gtfsrt.VehiclePosition_OccupancyStatus {
+	return &s
+}
```

Comment:
If it makes your life easier, now that Go has generics all of these functions could be replaced by a single one:

```
func ptr[T any](t T) *T {
  return &t
}
```

Reply:
Great suggestion, replaced all functions with a generic version.